### ISSUE TITLE
elif self.fsdp is not None and self.args.fsdp_config["xla"]:

### ISSUE THREAD

**Issue by duanzhenyu001** [huggingface/transformers#24705]
### System Info

- `transformers` version: 4.24.0
- Platform: Linux-5.4.143.bsk.8-amd64-x86_64-with-glibc2.28
- Python version: 3.10.9
- Huggingface_hub version: 0.10.1
- PyTorch version (GPU?): 1.12.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@sgugger

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

in trainer if self.args.fsdp_config["xla"] is true，trainer will wrap model layers，is this mean if i want to use fsdp to shard model，I must install torch-xla>2.0?

### Expected behavior

want to know if  i must install torch-xla for fspd trainning

**Comment by amyeroberts** [comment]
Hi @duanzhenyu001, thanks for raising an issue! 

This is a question best placed in our [forums](https://discuss.huggingface.co/). We try to reserve the github issues for feature requests and bug reports.

That being said, checking the docs it's possible to see that [XLA supports FDSP](https://huggingface.co/docs/transformers/main/main_classes/trainer#pytorchxla-fully-sharded-data-parallel), but [FDSP can be used separately](https://huggingface.co/docs/transformers/main/main_classes/trainer#pytorch-fully-sharded-data-parallel). If you wish to use XLA with FDSP, it's necessary to install torch-xla >= 2.0.

**Comment by duanzhenyu001** [comment]
@amyeroberts thanks a lot for your reply. when I use huggingface trainer fsdp mode finetune a 6B model，I got oom even 8 V100-32G gpu used. I didn't find where trainer shard my model when I use fsdp without xla. could you please point it out in the trainer code. thanks very much.

**Comment by duanzhenyu001** [comment]
@amyeroberts  and here is my TrainingArguments when I finetune glm2 a 6B model.

**Comment by duanzhenyu001** [comment]
@amyeroberts and in "create_accelerator_and_postprocess" function, 
self.accelerator = Accelerator(
            deepspeed_plugin=self.args.deepspeed_plugin,
            gradient_accumulation_steps=self.args.gradient_accumulation_steps,
        )
self.accelerator always be inited, self.is_fsdp_enabled = getattr(self.accelerator.state, "fsdp_plugin", None) is not None would always be False, I'm so confused by the purpose of this function.

**Comment by duanzhenyu001** [comment]
@amyeroberts and at function "create_accelerator_and_postprocess" , 
        self.accelerator = Accelerator(
            deepspeed_plugin=self.args.deepspeed_plugin,
            gradient_accumulation_steps=self.args.gradient_accumulation_steps,
        )
self.accelerator always be inited, if i don't set os env ACCELERATE_USE_FSDP to be true, "self.is_fsdp_enabled = getattr(self.accelerator.state, "fsdp_plugin", None) is not None"  self.is_fsdp_enabled will always be false,  is this what you realy wanted?  and I'm confused the intention of this function.

**Comment by duanzhenyu001** [comment]
> 

I'm now using v4.30.2，but still get oom memory errors when training on 8 V100-32g GPUs. @pacman100  can some one give any suggestions? thanks a lot

**Comment by pacman100** [comment]
The following works on 4 A100 80GB GPUs with GPT-j (6B model), although the entire VRAM is occupied on each:

```
cd transformers/examples/pytorch/language-modeling

torchrun --nnodes 1 --nproc-per-node 2 run_clm.py   --model_name_or_path EleutherAI/gpt-j-6b --dataset_name wikitext   --dataset_config_name wikitext-2-raw-v1   --per_device_train_batch_size 1   --per_device_eval_batch_size 1   --do_train   --do_eval   --output_dir /tmp/test-clm   --gradient_accumulation_steps 8 --overwrite_output_dir --fsdp "full_shard auto_wrap" --fsdp_transformer_layer_cls_to_wrap "GPTJBlock" --bf16
```

The reason for OOM at your end is plausible due to large seq lengths such as >=1024. In such cases, gradient/activation checkpointing is recommended.
